sanic/response.py

```diff
@@ -112,8 +114,16 @@ async def send(
         """
         if data is None and end_stream is None:
             end_stream = True
-        if end_stream and not data and self.stream.send is None:
-            return
+        if self.stream.send is None:
+            if end_stream and not data:
+                return
+            elif self.stream.stage == Stage.IDLE:
+                raise SanicException(
+                    "Response stream was ended, no more response data is"
+                    " allow ed to be sent."
+                )
+            else:
+                raise SanicException("Send response function isn't available")
```

Comment:
Should we add the `Stage` to this?

Reply:
I am not sure. I couldn't think about any other situation that the function is None, but I added a else just in case it falls to this.

Reply:
I agree that it is the correct approach to have a failsafe, but I was just thinking that if that happens maybe we should include a little bit of information about whatever unexpected state was reached.

Reply:
What info can be included do you think? I couldn't think about more than that.